src/olympia/abuse/models.py

```diff
@@ -1233,130 +1204,100 @@ def appeal(self, *, abuse_report, appeal_text, user, is_reporter):
-                'policies': policies if not self.notes else (),
-                'version_list': ', '.join(ver_str for ver_str in version_numbers),
-                'has_attachment': hasattr(log_entry, 'attachmentlog'),
-                'dev_url': absolutify(self.target.get_dev_url('versions'))
-                if self.addon_id
-                else None,
-            },
-        )
+            self.update(cinder_id=decision_cinder_id)
 
-    def process_action(self, *, release_hold=False):
-        """currently only called by decisions from cinder.
-        see https://mozilla-hub.atlassian.net/browse/AMOENG-1125
-        """
-        assert not self.action_date  # we should not be attempting to process twice
+    def execute_action_and_notify(self, *, release_hold=False):
+        """Execute the action for the decision, if not already carried out.
+        The action may be held for 2nd level approval.
+        If the action has been carried out, notify interested parties"""
+        action_helper = self.get_action_helper()
         cinder_job = self.originating_job
-        appealed_action = (
-            getattr(cinder_job.appealed_decisions.first(), 'action', None)
-            if cinder_job
-            else None
-        )
+        log_entry = None
+        if not self.action_date:
+            if release_hold or not action_helper.should_hold_action():
+                # We set the action_date because .can_be_appealed depends on it
+                self.action_date = datetime.now()
+                log_entry = action_helper.process_action()
+                # But only save it afterwards in case process_action failed
+                self.save(update_fields=('action_date',))
+            else:
+                log_entry = action_helper.hold_action()
 
-        overridden_action = (
-            self.override_of
-            and self.override_of.action_date
-            and self.override_of.action
-        )
-        action_helper = self.get_action_helper(
-            overridden_action=overridden_action,
-            appealed_action=appealed_action,
-        )
-        if release_hold or not action_helper.should_hold_action():
-            self.action_date = datetime.now()
-            log_entry = action_helper.process_action()
+        log_entry = log_entry or self.activities.first()
+
+        if self.action_date:
             if cinder_job:
                 cinder_job.notify_reporters(action_helper)
-            action_helper.notify_owners(log_entry_id=getattr(log_entry, 'id', None))
-            self.save(update_fields=('action_date',))
-        else:
-            action_helper.hold_action()
+
+            if self.addon_id:
+                details = (log_entry and log_entry.details) or {}
+                # Using 'reviewtype' is a bit of a hack to allow us to differentiate
+                # decisions from the reviewer tools, because it's only set there.
+                from_reviewer_tools = 'reviewtype' in details
```

Comment:
Don't love this... It's fairly fragile. Not that much better but could we use `human_review` instead ? `reviewtype` is kinda obsolete these days...

At the minimum add a comment in `ReviewBase.log_action()` to make sure we continue setting this in the future...

Reply:
mmm, forgot the comment.  I'll add it with https://github.com/mozilla/addons/issues/1982